DEVELOP.rst

```diff
@@ -1,20 +1,122 @@
 .. highlight:: sh
 
-===============
-Developer Guide
-===============
+###############
+Developer guide
+###############
 
-API Spec
-========
 
-See the `API spec`_ document for more info.
+*****
+About
+*****
 
-Integration Tests
+This section of the documentation describes the API specification document, the
+corresponding test cases, and walks you through the process of provisioning a
+CrateDB sandbox environment using Docker, and running the scenario on behalf
+of the Python backend application, as an example.
+
+API specification
 =================
 
-Run the integration tests::
+The HTTP API of all the applications hosted within this repository is described
+within the `Backend API specification for the CrateDB guestbook demo application`_
+document.
+
+
+
+***********
+Walkthrough
+***********
+
+
+Docker setup
+============
+
+If you choose to use Docker for running CrateDB on your workstation, use those
+commands to run and provision your database instance::
+
+    # Define CrateDB version.
+    export CRATEDB_VERSION=latest
+    export CRATEDB_IMAGE=crate:${CRATEDB_VERSION}
+
+    # Start CrateDB.
+    docker run -it --rm \
+        --volume=$(pwd)/sql:/sql \
+        --publish=4200:4200 --publish=5432:5432 \
+        ${CRATEDB_IMAGE} \
+        -Cdiscovery.type=single-node \
+        -Ccluster.routing.allocation.disk.threshold_enabled=false
+
+    # Populate schema and data, and verify it's there.
+    docker run --rm --network=host --volume=$(pwd)/sql:/sql ${CRATEDB_IMAGE} sh -c "$(cat <<EOT
+        crash < /sql/schemas.sql;
+        crash <<SQL
+            COPY guestbook.countries FROM 'file:///sql/countries.json' RETURN SUMMARY;
+            REFRESH TABLE guestbook.countries;
+            SELECT id, name FROM guestbook.countries LIMIT 10;
```

Comment:
> It may even be wrapped up into a little helper program for that very purpose.

Probably nothing more than:
```
COUNTRIES_URL=$(rclone sync /sql/countries.json s3:temp/sql/countries.json)
```
```sql
COPY guestbook.countries FROM '${COUNTRIES_URL}' RETURN SUMMARY;
```

Reply:
This approach is smart, but an `rclone` config file would need to be in place

Reply:
Oh well. Actually, I didn't use `rclone` any time before. Now, I also discovered that it insists on creating a configuration file beforehand. Bummer. This can't be true?

Reply:
It looks like `rclone` can also be configured with environment variables instead of a configuration file, see [^1].

> For example, to configure [...] _without a config file_, use the unix way of setting environment variables.

[^1]: https://rclone.org/docs/#config-file